packages/discord.js/src/structures/MessagePayload.js

```diff
@@ -134,8 +135,18 @@ class MessagePayload {
     }
 
     const enforce_nonce = Boolean(this.options.enforceNonce);
-    if (enforce_nonce && nonce === undefined) {
-      throw new DiscordjsError(ErrorCodes.MessageNonceRequired);
+
+    /**
+     * If `nonce` isn't provided, generate one & set `enforceNonce`
+     * Unless `enforceNonce` is explicitly set to `false`(not just falsy)
+     */
```

Comment:
I don't think this should be a JSDoc 
```suggestion
    // If `nonce` isn't provided, generate one & set `enforceNonce`
    // Unless `enforceNonce` is explicitly set to `false`(not just falsy)
```